libbeat/cfgfile/cfgfile.go

```diff
@@ -29,12 +29,13 @@ func ChangeDefaultCfgfileFlag(beatName string) error {
 		return fmt.Errorf("Flag -c not found")
 	}
 
-	if runtime.GOOS == "windows" {
-		cliflag.DefValue = fmt.Sprintf(`C:\Program Files\%s\%s.yml`,
-			strings.Title(beatName), beatName)
-	} else {
-		cliflag.DefValue = fmt.Sprintf("/etc/%s/%s.yml", beatName, beatName)
+	path, err := filepath.Abs(filepath.Dir(os.Args[0]))
+	if err != nil {
+		fmt.Print("Could not detect path of executable.")
```

Comment:
I think this function should propagate the error since you cannot use `path` afterwards. Instead of printing, use something like

`return fmt.Errorf("Failed to set default config file location because the absolute path to %s could not be obtained. %v", os.Args[0], err)`